### ISSUE TITLE
training on a custom dataset

### ISSUE THREAD

**Issue by orange2xiaoyu** [sconlyshootery/FeatDepth#34]
Hi, thank you so much for sharing this great project!

I have trained the model using my own datasets, and the images size of my trained datasets is (1280,720). 
In the first training, I trained my own datasets using input size of (512,288), and I got the results as figure 2.
In the second training,  I trained using input size of (1024,576), I only got a black image after 10epoch shown as figure 3，and I found the loss is not converge shown as figure 4.

I have only modified the input size in config file between the first training and the second training because the input camera intrinsics is normalization.
Do you have an idea what is going?

Thank you very much for your help!

![error](https://user-images.githubusercontent.com/49177034/106411423-45870480-6480-11eb-8ba9-aa492771d83d.png)

**Comment by orange2xiaoyu** [comment]
Hi, I found another problem with the following code:
![net_error](https://user-images.githubusercontent.com/49177034/106549463-f4444700-654b-11eb-969f-df39798cc127.png)


I want to know whether this parameter is fixed, and should it be changed to the same size as the input size as shown?
`pose_feats = {f_i: F.interpolate(inputs["color_aug", f_i, 0], [self.opt.height, self.opt.width], mode="bilinear", align_corners=False) for f_i in self.opt.frame_ids}`

**Comment by sconlyshootery** [comment]
(1) If you change the input image size, you should revise camera intrinsices.
For exmaple, denotes camera intrinsices is K which is a 3*3 matrix,
K[0] *= target_image_width / original_img_width
K[1] *= target_image_height / original_img_height
(2) the self.opt.height and self.opt.width is the image size for posenet, you should keep the same aspect ratio.

**Comment by orange2xiaoyu** [comment]
Thank you for your replay.

(1) I am sorry, I don't quite understand where this camera intrinsices should be modified. In this code, the camera intrinsics matrix k is normalization as following, 
![data_error](https://user-images.githubusercontent.com/49177034/106717485-c2f67480-663a-11eb-84cb-241cf11e7653.png)

and in mono_dataset.py, camera intrinsics matrix can be obtained using the input size,
K[0, :] *= self.width
K[1, :] *= self.height
Is this the parameter `K` that needs to be modified?

(2) I have modified the image size(1024,320) to (512,288) in cfg_kitti_fm.py for training our datasets, whether the self.opt.height and self.opt.width should be replaced (192, 640) in posenet code?

**Comment by sconlyshootery** [comment]
(1) It is correct if you normalize the camera intrinsics before writing into cfg files.
(2) right, you can also use a smaller size, only if you keep the same aspect ratio.

I think the problem may caused by static camera, since the whole method is built on structure from motion theory, if no motion exist between continuous frames, it will make the whole scene is from a infinite distance.

**Comment by buaachen1993** [comment]
@orange2xiaoyu hi, I meet the same problem using my own data, have you find a resolution? Thanks a lot!

**Comment by orange2xiaoyu** [comment]
Do you mean the first issue？ 
Maybe you load the kitti pretrained model before training?
But I haven't tested whether the problem will be solved if do not loading the kitti pretrained model,  maybe you can try it.

**Comment by buaachen1993** [comment]
> Do you mean the first issue？
> Maybe you load the kitti pretrained model before training?
> But I haven't tested whether the problem will be solved if do not loading the kitti pretrained model, maybe you can try it.

I mean i got  black image too using my own data，even if i removed the static images. Also, i used the imagenet pretrained weight instead of the kitti pretrained model.
Is your result still total black?

**Comment by orange2xiaoyu** [comment]
Are you sure you get the correct  camera intrinsics for your own data? 
And whether the input size keeps the same aspect ratio with your original image size?

**Comment by buaachen1993** [comment]
> Are you sure you get the correct camera intrinsics for your own data?
> And whether the input size keeps the same aspect ratio with your original image size?

1. I don‘t have the exact camera intrinsics for my own data, this might be the reason。
2. Yes, I do keep the aspect ratio.

Thank you for your timely reply, I will continue to try to solve this problem.

**Comment by cocacola0** [comment]
Hello @orange2xiaoyu. Did you find any solutions to remove black images when training on certain image sizes? I have tried multiplying the intrinsic parameter as follows: 
K[0, :] *= self.current_width/self.original_width
K[1, :] *= self.current_height/self.original_height
More than that, I switched the hardcoded input size in pose_feats from [192, 640] to [self.opt.height, self.opt.width]. With all of this being done, some image sizes work, and some don't. What solutions did you find in your case? Thank you in advance!

**Comment by sconlyshootery** [comment]
https://github.com/sconlyshootery/FeatDepth/issues/40#issue-816588356
This issue said that when the width and height are multiples of 64 works, you can try whether this assumption is correct.
I will also check my code to figure it out.

**Comment by sconlyshootery** [comment]
mono/datasets/folder_dataset.py
BTW, you can refer to this code to build your own dataloader for your custom dataset.

And you can show me your codes about dataloader, I will help you review it.

**Comment by orange2xiaoyu** [comment]
This code for my custom dataset reference to kitti_dataset.py.
[navinfo_dataset.txt](https://github.com/sconlyshootery/FeatDepth/files/6047696/navinfo_dataset.txt)

**Comment by sconlyshootery** [comment]
https://github.com/sconlyshootery/FeatDepth/blob/db7344ae9d9cefafc92983fc8c5b30920751be68/mono/datasets/mono_dataset.py#L152
Whether your image name is continuous, here the dataloader will find frame_index + i, if it fails, it will still load current image.

**Comment by orange2xiaoyu** [comment]
Of course, my image names are continuous.
And I can get correct depth image using input size of (512,288).

**Comment by sconlyshootery** [comment]
Someone tell me that the issue can be solved in this way, 'Later I figured it out that there was nothing wrong with the code. Due to a change in nvidia-driver version, I was facing this issue. I was able to solve this issue by installing numba==0.43.1. ’
You may try.

**Comment by orange2xiaoyu** [comment]
Ok, thank you very much!

**Comment by sconlyshootery** [comment]
I will close this issue, you can reopen it if you have further question.
